### ISSUE TITLE
Nested CTEs (WITH statements) fail to parse

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#338]
BigQuery (and possibly other SQL dialects) allow `WITH statements to be nested one inside another. Here's an example adapted from one of our projects. This does not parse. If I reorganize the code, placing the nested `raw` CTE at the same level as `clean`, the query parses with no issues.

```
WITH
  clean AS (
  WITH
    raw AS (
    SELECT
      table_c.user_id,
      table_c.campaign_id
    FROM
      table_c
    JOIN
      table_cs
    USING
      (user_id,
        campaign_id)
    )
  SELECT
    user_id,
    campaign_id
  FROM
    raw
  )

SELECT * FROM clean
```

**Comment by alanmcruickshank** [comment]
👍 . I've just done a reorg of this part of the codebase *anyway* so should be easy to add in at the same time.

**Comment by alanmcruickshank** [comment]
this was fixed in #409
